deadseeker/deadseeker.py

```diff
@@ -0,0 +1,111 @@
+from .common import (
+    SeekerConfig,
+    UrlFetchResponse,
+    UrlTarget,
+    SeekResults
+)
+
+logger = logging.getLogger(__name__)
+
+
+class DeadSeeker:
+    def __init__(self, config: SeekerConfig) -> None:
+        self.config = config
+
+    async def _get_urlfetchresponse(
+            self,
+            session: aiohttp.ClientSession,
+            urltarget: UrlTarget) -> UrlFetchResponse:
+        resp = UrlFetchResponse(urltarget)
+        start = time.time()  # measure load time (HEAD only)
+        url = urltarget.url
+        end: float = -1
+        try:
+            async with session.head(url) as headresponse:
+                end = time.time()
+                resp.status = headresponse.status
+                has_html = 'html' in headresponse.headers['Content-Type']
+                onsite = urltarget.home in url
+                if(has_html and onsite):
+                    async with session.get(url) as getresponse:
+                        resp.html = await getresponse.text()
+        except aiohttp.ClientResponseError as e:
+            resp.status = e.status
+            resp.error = e
+        except aiohttp.ClientError as e:
+            resp.status = -1
+            resp.error = e
+        if end < 0:
+            end = time.time()
+        resp.elapsed = (end - start)*1000
+        return resp
+
+    async def _main(self, urls: List[str]) -> SeekResults:
+        start = time.time()
+        results = SeekResults()
+        visited: Set[str] = set()
+        targets: Deque[UrlTarget] = Deque()
+        for url in urls:
+            visited.add(url)
+            targets.appendleft(UrlTarget(url, url, self.config.max_depth))
+        linkparser = LinkParser(self.config.linkacceptor)
+        async with createClientSession(self.config) as session:
+            while targets:
+                tasks = []
+                while targets:
+                    urltarget = targets.pop()
+                    tasks.append(
+                        asyncio.create_task(
+                            self._get_urlfetchresponse(session, urltarget)))
+                for task in asyncio.as_completed(tasks):  # completed first
+                    resp = await task
+                    self._log_result(resp)
+                    if(resp.error):
+                        results.failures.append(resp)
+                    else:
+                        results.successes.append(resp)
+                    url = resp.urltarget.url
+                    depth = resp.urltarget.depth
+                    if(resp.html and depth != 0):
+                        home = resp.urltarget.home
+                        linkparser.reset()
+                        linkparser.feed(resp.html)
+                        for newurl in linkparser.links:
+                            if not bool(
+                                    urlparse(newurl).netloc):  # relative link?
+                                newurl = urljoin(resp.urltarget.home, newurl)
```

Comment:
I think the error with wrong subdirectory links is caused by this line

Reply:
Great test case!   You're right, the problem is that I'm using the original resp.urltarget.home value, but it appears to work if I just use the url for the main page.  I don't think we actually need the "home" field in the UrlTarget class anymore

Reply:
Oops, I take that back, the "home" field is used to see if the url belongs to the original site.  But I think this fixes the issue, but the logging test still fails:
```suggestion
                                newurl = urljoin(url, newurl)
```